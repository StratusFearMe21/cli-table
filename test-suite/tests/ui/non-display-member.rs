//! All the members of a struct should implement `Display` (other option is to use `display_fn`)
use cli_table::Table;

#[derive(Table)]
struct Test {
    #[table(title = "a")]
    a: Vec<u8>,
}

fn main() {}
