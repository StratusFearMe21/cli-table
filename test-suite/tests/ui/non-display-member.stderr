error[E0277]: `Vec<u8>` doesn't implement `std::fmt::Display`
 --> $DIR/non-display-member.rs:6:5
  |
6 | /     #[table(title = "a")]
7 | |     a: Vec<u8>,
  | |______________^ `Vec<u8>` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `Vec<u8>`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required because of the requirements on the impl of `std::fmt::Display` for `&Vec<u8>`
  = note: required because of the requirements on the impl of `cli_table::Cell` for `&Vec<u8>`
  = note: required by `cell`
